src/GazeboThruster.cpp

```diff
@@ -192,6 +192,7 @@ void GazeboThruster::updateBegin(common::UpdateInfo const& info)
             thruster != thrusters.end(); ++thruster)
     {
         physics::LinkPtr link = model->GetLink( thruster->name );
+        link->AddLinkForce( thruster->effort*gazebo::math::Vector3::UnitX);
         link->AddLinkForce( thruster->effort*thruster->direction, thruster->position );
```

Comment:
> Wouldn't it be less convoluted to transfer the thruster force to the CoG and then apply the added mass ?

I could, but then I'd need to compute the torque. Also, the efforts need to be applied in the thruster frame once they are applied in the thruster link.
Applying the effort in the computed direction and position guaranties that the torque are also right.

The direction and position are computed as:
[f, t] = [`R`\*`ft`, (`pt`-cog).cross(`R`*`ft`)]
`ft` is the force applied by the thruster, `R` is the rotation from thruster frame to body frame, `pt` is the position of the thruster frame.

If I multiple the it by a matrix M, i would have have a new force and toque [f', t'] in the body frame. Then I convert the force to the thruster frame, giving the direction, and find the position that determine the torque.
Like that just one force is applied. The direction and position guarantees that both the force and torque are right.

>I'm missing something 

Direction is not UnitX and the norm may not be 1. Position is not the (0,0,0) of thruster frame.
For Ma=0 => direction=0, and just the original force is considered.